packages/integrations/cloudflare/src/server.directory.ts

```diff
@@ -10,25 +10,16 @@ if (!isNode) {
 export function createExports(manifest: SSRManifest) {
 	const app = new App(manifest);
 
-	const onRequest = async ({
-		request,
-		next,
-		...runtimeEnv
-	}: {
-		request: Request & CFRequest;
-		next: (request: Request) => void;
-		waitUntil: EventContext<unknown, any, unknown>['waitUntil'];
-	} & Record<string, unknown>) => {
-		process.env = runtimeEnv.env as any;
+	const onRequest = async (context: EventContext<unknown, string, unknown>) => {
+		const request = context.request as CFRequest & Request
+		const { next, env } = context
+
+		process.env = env as any; // would love to remove this any cast in the future
 
 		const { pathname } = new URL(request.url);
 		// static assets fallback, in case default _routes.json is not used
 		if (manifest.assets.has(pathname)) {
-			// we need this so the page does not error
-			// https://developers.cloudflare.com/pages/platform/functions/advanced-mode/#set-up-a-function
-			return (runtimeEnv.env as EventContext<unknown, string, unknown>['env']).ASSETS.fetch(
-				request
-			);
+			return env.ASSETS.fetch(request)
```

Comment:
This is actually a bug fix, which otherwise breaks CF. See `[server.advanced.ts](https://github.com/withastro/astro/pull/7541/files#diff-854537ee160391dda208e76dbea4c1471189e0368d2d76537485a0039d103da3)`, where this fix was already introduced weeks ago.